### DISCUSSION TITLE
Empty Build Event Macros

### DISCUSSION THREAD

**Discussion by cubt85iz** [wixtoolset/issues#7393] [Questions] [answered]
I'm attempting to execute a Powershell script as part of the Pre-build for a new WiX 4 MSI Package using the following command:

`powershell.exe -ExecutionPolicy Bypass -NoProfile -NonInteractive -File pre-build.ps1 $(ProjectDir)`

It appears that the ProjectDir macro is empty. I'm using Heatwave for VS2022 1.0.0 and I see that #7020 has been closed.

Am I missing something here or is this an issue that needs to be written up?

Thanks 😁

**Comment by robmen** [comment, marked as answer]
You need to manually import `Sdk.targets` and move your pre-build event after the import. Someone else asked this a bit ago. Someone should probably add it to the migration FAQ.

**Comment by cubt85iz** [comment]
Thanks for the hint. For those who may stumble upon this discussion, here's exactly what I needed to do.

1. Open the wixproj for editing.
2. Remove Sdk from Project to prevent duplicate import warning (as noted [here](https://learn.microsoft.com/en-us/visualstudio/msbuild/how-to-use-project-sdk?view=vs-2022#use-the-import-element-anywhere-in-your-project))
3. Add import statements for Sdk.props and Sdk.targets after the Project tag.

Here is an excerpt from my file now.

```
<Project>
  <Import Project="Sdk.props" Sdk="WixToolset.Sdk" Version="4.0.0" />
  <Import Project="Sdk.targets" Sdk="WixToolset.Sdk" Version="4.0.0" />
  <PropertyGroup>
    <PreBuildEvent>:: powershell.exe -ExecutionPolicy Bypass -NoProfile -NonInteractive -File pre-build.ps1 $(ProjectDir)
			echo Test $(ProjectDir)</PreBuildEvent>
    <Platforms>x64;x86;ARM64</Platforms>
  </PropertyGroup>
```

**Reply by robmen** [reply]
@cubt85iz why not take this knowledge you found and find the right page on https://wixtoolset.org/docs/ and add it there? Every page there has a link to edit the page. Then you're proactively helping the next person.

**Reply by cubt85iz** [reply]
Roger that. I had it on my todo list to figure that out.

BTW thanks for the excellent live streams. They really helped me hit the ground running with WiX 4.
